provisionerd/provisionerd.go

```diff
@@ -733,6 +748,60 @@ func (p *Server) runTemplateImportProvision(ctx, shutdown context.Context, provi
 	}
 }
 
+func (p *Server) runTemplatePlan(ctx, shutdown context.Context, provisioner sdkproto.DRPCProvisionerClient, job *proto.AcquiredJob) {
+	// Ensure all metadata fields are set as they are all optional for plans.
+	metadata := job.GetTemplatePlan().GetMetadata()
+	metadata.WorkspaceTransition = sdkproto.WorkspaceTransition_START
+	if metadata.CoderUrl == "" {
+		metadata.CoderUrl = "http://localhost:3000"
+	}
```

Comment:
*minor:* Can we get this from an environment variable or something instead?

Reply:
It doesn't seem easily accessible, but also shouldn't have much bearing on the test since this is only used for a plan. No connections should be made to this URL AFAIK. We do set this from coderd when we create the job though, so in practice this if condition will never be hit and only exists for extra safety